eval/eval.py

```diff
@@ -323,155 +351,262 @@ def add_sentence(
+                print(f"   {c:<13}:           {self._sentences[c]['num_tokens']:6}")
+            # Total number of sentences processed
+            print(f"\nSentences processed:        {num_sentences:6}")
+            for c in CATEGORIES:
+                print(f"   {c:<13}:           {self._sentences[c]['count']:6}")
+
+        def perc(n, whole):
             """ Return a percentage of total sentences, formatted as 3.2f """
-            if num_sentences == 0:
+            if whole == 0:
                 return "N/A"
-            return f"{100.0*n/num_sentences:3.2f}"
-
-        # Total number of true negatives found
-        true_negatives = sum(d["true_negatives"] for d in self._sentences.values())
-        print(
-            f"\nTrue negatives:             {true_negatives:6} {perc(true_negatives):>6}%"
-        )
-        for c in CATEGORIES:
-            print(f"   {c:<13}:           {self._sentences[c]['true_negatives']:6}")
-
-        # Total number of true positives found
-        true_positives = sum(d["true_positives"] for d in self._sentences.values())
-        print(
-            f"\nTrue positives:             {true_positives:6} {perc(true_positives):>6}%"
-        )
-        for c in CATEGORIES:
-            print(f"   {c:<13}:           {self._sentences[c]['true_positives']:6}")
-
-        # Total number of false negatives found
-        false_negatives = sum(d["false_negatives"] for d in self._sentences.values())
-        print(
-            f"\nFalse negatives:            {false_negatives:6} {perc(false_negatives):>6}%"
-        )
-        for c in CATEGORIES:
-            print(f"   {c:<13}:           {self._sentences[c]['false_negatives']:6}")
-
-        # Total number of false positives found
-        false_positives = sum(d["false_positives"] for d in self._sentences.values())
-        print(
-            f"\nFalse positives:            {false_positives:6} {perc(false_positives):>6}%"
-        )
-        for c in CATEGORIES:
-            print(f"   {c:<13}:           {self._sentences[c]['false_positives']:6}")
-
-        # Percentage of true vs. false
-        true_results = true_positives + true_negatives
-        false_results = false_positives + false_negatives
-        if num_sentences == 0:
-            result = "N/A"
-        else:
-            result = f"{100.0*true_results/num_sentences:3.2f}%/{100.0*false_results/num_sentences:3.2f}%"
-        print(f"\nTrue/false split: {result:>16}")
-        for c in CATEGORIES:
-            d = self._sentences[c]
-            num_sentences = d["count"]
-            true_results = d["true_positives"] + d["true_negatives"]
-            false_results = d["false_positives"] + d["false_negatives"]
-            if num_sentences == 0:
+            return f"{100.0*n/whole:3.2f}"
+        
+        def write_basic_value(val, bv, whole, errwhole=None) -> None:
+            """ Write basic values for sentences and their freqs to stdout """
+            if errwhole and errwhole != 0:
```

Comment:
Probably enough to say `if errwhole:`  - or, for slightly different functionality, `if errwhole is not None:`

Reply:
Done!